src/gui/sdlmain.cpp

```diff
@@ -1216,15 +1221,18 @@ static void notify_new_mouse_screen_params()
 	params.draw_rect =
 	        to_rect(sdl.draw_rect_px).Copy().Scale(1.0f / sdl.desktop.dpi_scale);
 
-	int abs_x = 0;
-	int abs_y = 0;
+	// TODO: SDL 3 has changed this function to return floats for sub-pixel precision
+	// Maybe update the member variables to float and take advantage of this?
+	float abs_x = 0;
+	float abs_y = 0;
 	SDL_GetMouseState(&abs_x, &abs_y);
 
-	params.x_abs = check_cast<int32_t>(abs_x);
-	params.y_abs = check_cast<int32_t>(abs_y);
+	params.x_abs = static_cast<int32_t>(abs_x);
```

Comment:
For now I suggest to put `iroundf(abs_x)`.

Indeed, the mouse code will have to be adapted, but this might require some deeper modifications to fully utilize the sub-pixel precision - I haven't foreseen such API enhancement.

Reply:
Do we _need_ subpixel precision?

Reply:
Maybe not, but we would better handle it correctly before our first SDL 3 release, or bad things might happen.

Imagine someone gets a _Super-Duper 8 kHz Ultra-Gaming AI Mouse+_ and SDL starts throwing at us mouse movement values like:

0.31, 0.85, 1.63, 0.75, 0.42, 0.37, 0.45, ...

Rounding won't help and mouse movement is going to be rather jerky in such case.

BTW, testing the updated mouse code is going to be a great opportunity to prepare the mouse testing guidelines on our wiki :D

Reply:
The VMware/VirtualBox mouse emulation code, in order to perform the necessary calculations, converts the integer absolute position to... floats :) So definitely there is no point in rounding the values in `sdlmain.c`.

Reply:
Ok, if it's easy and has benefits, go for it. I guess I'm saying, just be pragmatic. Just because feature X is supported by SDL, we don't _have_ to use it, unless it brings some objective improvements.

But, probably an easy change, like you said.

Reply:
Better than on current SDL 2.x branch? Maybe on some platforms you'll get a slightly more precise mouse cursor if using seamless mouse integration. But I'm not sure how many people will notice.

Better than on current SDL 3.x branch? If you are playing Civilization on Windows 3.1 using a wheel-enabled mouse driver (like the A4Tech one) - then possibly (again, depends on the platform, mouse, etc.). If SDL 3 starts sending us non-integer wheel movement values (which is now possible, with the new API), the experience won't be smooth, it might even make the wheel non-working.

Reply:
I got a question related to mice in relationship to the mouse discussion. I have a huion touchscreen tablet and in Deluxe Paint II work it exactly as like it's a mouse with proper precision pointing where the stylus is. It's very funky in other programs like Autodesk Animator Pro, someone told me in the github repo for AAP that it appears the program uses deltas like delta x and delta y instead of absolute positioning. How will SDL3 improve this regards? I'm curious to know.

Reply:
On the main brach the mouse emulation API now accepts float - so no need for any rounding after rebase :)

Reply:
Wtf is up with these "ghost" users? People register on GitHub, make two comments, then delete their accounts??? I've seen the same pattern before. Weird.